packages/cache/src/stylis-plugins.js

```diff
@@ -1,89 +1,47 @@
-// @flow
-// https://github.com/thysultan/stylis.js/tree/master/plugins/rule-sheet
-// inlined to avoid umd wrapper and peerDep warnings/installing stylis
-// since we use stylis after closure compiler
+const last = arr => (arr.length ? arr[arr.length - 1] : null)
 
-import type { StylisPlugin } from './types'
-
-const delimiter = '/*|*/'
-const needle = delimiter + '}'
-
-function toSheet(block) {
-  if (block) {
-    Sheet.current.insert(block + '}')
+export let removeLabel = element => {
+  if (element.type === 'decl') {
+    var value = element.value
+    if (
+      // charcode for l
+      value.charCodeAt(0) === 108 &&
+      // charcode for b
+      value.charCodeAt(2) === 98
+    ) {
+      // this ignores label
+      element.return = ''
+      element.value = ''
+    }
   }
 }
 
-export let Sheet: { current: { +insert: string => void } } = {
-  current: (null: any)
-}
+const ignoreFlag =
+  'emotion-disable-server-rendering-unsafe-selector-warning-please-do-not-use-this-the-warning-exists-for-a-reason'
 
-export let ruleSheet: StylisPlugin = (
-  context,
-  content,
-  selectors,
-  parents,
-  line,
-  column,
-  length,
-  ns,
-  depth,
-  at
-) => {
-  switch (context) {
-    // property
-    case 1: {
-      switch (content.charCodeAt(0)) {
-        case 64: {
-          // @import
-          Sheet.current.insert(content + ';')
-          return ''
-        }
-        // charcode for l
-        case 108: {
-          // charcode for b
-          // this ignores label
-          if (content.charCodeAt(2) === 98) {
-            return ''
-          }
-        }
-      }
-      break
-    }
-    // selector
-    case 2: {
-      if (ns === 0) return content + delimiter
-      break
-    }
-    // at-rule
-    case 3: {
-      switch (ns) {
-        // @font-face, @page
-        case 102:
-        case 112: {
-          Sheet.current.insert(selectors[0] + content)
-          return ''
-        }
-        default: {
-          return content + (at === 0 ? delimiter : '')
-        }
-      }
-    }
-    case -2: {
-      content.split(needle).forEach(toSheet)
-    }
-  }
-}
+const isIgnoringComment = element =>
+  !!element &&
+  element.type === 'comm' &&
+  element.children.indexOf(ignoreFlag) > -1
 
-export let removeLabel: StylisPlugin = (context, content) => {
-  if (
-    context === 1 &&
-    // charcode for l
-    content.charCodeAt(0) === 108 &&
-    // charcode for b
-    content.charCodeAt(2) === 98
-    // this ignores label
-  ) {
-    return ''
+export let createUnsafeSelectorsAlarm = cache => (element, index, children) => {
```

Comment:
Could you check that this is dead code eliminated in prod?

Reply:
sure, I'll recheck this later, but i don't see any reason why it wouldnt:
1. it's a declaration (side-effect freeeee)
2. it's called conditionally